### PR TITLE
Reenable analyzers

### PR DESCRIPTION
**Pull request by 333fred**
This reenables analyzers in this repo by using `PackageId` to work around the nuget cyclic dependency bug. Thanks to @sharwell for pointing this out. Doing this in favor of https://github.com/dotnet/roslyn-analyzers/pull/1356 as this is simpler and involves no renaming.

### CODE REVIEW

File: src/MetaCompilation.Analyzers/Core/MetaCompilation.Analyzers.csproj [hunk 1]
```diff
@@ -2,8 +2,13 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netstandard1.3</TargetFramework>
+    <!--
+      PackageId is used by Restore. If we set it to MetaCompilation.Analyzers,
+      Restore would conclude that there is a cyclic dependency between us and the MetaCompilation.Analyzers nuget package.
+    -->
+    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
```
Comment (mavasani) [inline src/MetaCompilation.Analyzers/Core/MetaCompilation.Analyzers.csproj]: Ah, good to know this is tricking NuGet. Do we know the NuGet bug for the cyclic dependency issue?

File: src/Directory.Build.props [hunk 2]
```diff
@@ -8,11 +8,17 @@
     <NoWarn>$(NoWarn);xUnit1004;xUnit2000;xUnit2003;xUnit2004;xUnit2009;xUnit2010;xUnit1013</NoWarn>
 
     <DefineConstants Condition="'$(USE_INTERNAL_IOPERATION_APIS)' == 'true'">$(DefineConstants),USE_INTERNAL_IOPERATION_APIS,DEFAULT_SEVERITY_SUGGESTION</DefineConstants>
-    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)Analyzers.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
   <!-- Test runner configuration -->
   <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.0-beta1-62314-01" />
```
Comment (mavasani) [inline src/Directory.Build.props]: Move the version string into Version.props.

File: src/MetaCompilation.Analyzers/Core/MetaCompilation.Analyzers.csproj (hunk 1 above)
Comment (333fred) [reply src/MetaCompilation.Analyzers/Core/MetaCompilation.Analyzers.csproj]: Looks like it's a won't fix: https://github.com/NuGet/Home/issues/936

File: src/Directory.Build.props (hunk 2 above)
Comment (333fred) [reply src/Directory.Build.props]: Fixed.

**Comment by 333fred** [comment]
Ping @tmat for a review please.

File: src/MetaCompilation.Analyzers/Core/MetaCompilation.Analyzers.csproj (hunk 1 above)
Comment (tmat) [reply src/MetaCompilation.Analyzers/Core/MetaCompilation.Analyzers.csproj]: I think we should reopen the issue. I don't think nuget team didn't consider it with analyzers in mind.

---
In reply to: [150998449](https://github.com/dotnet/roslyn-analyzers/pull/1358#discussion_r150998449) [](ancestors = 150998449)

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/Setup/FxCopAnalyzersPackage.cs [hunk 3]
```diff
@@ -7,8 +7,14 @@
 namespace FxCopAnalyzersSetup
 {
     [Guid(PackageGuid)]
+    // Package is used by VS
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes
```
Comment (tmat) [inline src/Microsoft.CodeAnalysis.FxCopAnalyzers/Setup/FxCopAnalyzersPackage.cs]: >#pragma warning disable CA1812 // Avoid uni [](start = 0, length = 43)

This is very messy and cluttered. I can't see the actual code anymore. Can you move the pragmas to the top of the file?

```C#
#pragma warning disable CA1812 
#pragma warning disable CA1823
```

There is no need to need to restoring these.

Comment (333fred) [reply src/Microsoft.CodeAnalysis.FxCopAnalyzers/Setup/FxCopAnalyzersPackage.cs]: Done.

**Pull request merged**